File: Front/src/app/models/book-request.ts
export interface BookRequest {
  id: string;
  userId: string;
  userFullName: string;
  title: string;
  author: string;
  reason: string;
  status: 'Pending' | 'Approved' | 'Rejected';
  adminMessage: string;
  createdAt: string;
}

File: Back/Back.Domain/Models/BookRequest.cs
using Supabase.Postgrest.Attributes;
using Supabase.Postgrest.Models;

namespace Back.Domain.Models;

[Table("BookRequests")]
public class BookRequest : BaseModel
{
    [PrimaryKey]
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string UserId { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Reason { get; set; } = string.Empty;
    public string Status { get; set; } = "Pending";
    public string AdminMessage { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}